src/lmstudio/json_api.py

```diff
@@ -1104,12 +1115,34 @@ class ToolFunctionDef:
 
     name: str
     description: str
-    parameters: Mapping[str, Any]
+    parameters: Mapping[str, type[Any] | ToolParamDefDict[Any]]
     implementation: Callable[..., Any]
 
+    def _extract_type_and_default(self, param_name: str, param_value: type[Any] | ToolParamDefDict[Any]) -> tuple[type[Any], Any | None]:
```

Comment:
`None` is a potentially valid default value, so this extraction needs to use a dedicated sentinel value to indicate the "no default defined" case. `_NO_DEFAULT = object()` will suffice for that, since it isn't a public sentinel, just an internal one.